kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/client/DefaultJsonHandlerSuite.scala

```diff
@@ -15,18 +15,204 @@
+      3,
+      TestRow("foo", "", null)
+    )
+  }
+
+  test("parse decimal type") {
+    testJsonParserWithSchema(
+      jsonString = """
+      |{
+      |  "col1":0,
+      |  "col2":0.01234567891234567891234567891234567890,
+      |  "col3":123456789123456789123456789123456789,
+      |  "col4":1234567891234567891234567891.2345678900,
+      |  "col5":1.23
+      |}
+      |""".stripMargin,
+      schema = new StructType()
+        .add("col1", DecimalType.USER_DEFAULT)
+        .add("col2", new DecimalType(38, 38))
+        .add("col3", new DecimalType(38, 0))
+        .add("col4", new DecimalType(38, 10))
+        .add("col5", new DecimalType(5, 2)),
+      TestRow(
+        new JBigDecimal(0),
+        new JBigDecimal("0.01234567891234567891234567891234567890"),
+        new JBigDecimal("123456789123456789123456789123456789"),
+        new JBigDecimal("1234567891234567891234567891.2345678900"),
+        new JBigDecimal("1.23")
+      )
+    )
+  }
+
+  test("parse date type") {
+    testJsonParserForSingleType(
+      jsonString = """{"col1":"2020-12-31"}""",
+      dataType = DateType.DATE,
+      1,
+      TestRow(18627)
+    )
+  }
+
+  test("parse timestamp type") {
+    testJsonParserForSingleType(
+      jsonString =
+        """
+          |{
+          | "col1":"2050-01-01T00:00:00.000-08:00",
+          | "col2":"1970-01-01T06:30:23.523Z"
+          | }
+          | """.stripMargin,
+      dataType = TimestampType.TIMESTAMP,
+      numColumns = 2,
+      TestRow(2524636800000000L, 23423523000L)
+    )
+  }
+
+  test("parse null input") {
+    val schema = new StructType()
+      .add("nested_struct", new StructType().add("foo", IntegerType.INTEGER))
+
+    val batch = jsonHandler.parseJson(
+      singletonStringColumnVector(null),
+      schema,
+      Optional.empty()
+    )
+    assert(batch.getColumnVector(0).getChild(0).isNullAt(0))
+  }
+
+  test("parse NaN and INF for float and double") {
+    def testSpecifiedString(json: String, output: TestRow): Unit = {
+      testJsonParserWithSchema(
+        jsonString = json,
+        schema = new StructType()
+          .add("col1", FloatType.FLOAT)
+          .add("col2", DoubleType.DOUBLE),
+        output
+      )
+    }
+    testSpecifiedString("""{"col1":"NaN","col2":"NaN"}""", TestRow(Float.NaN, Double.NaN))
+    testSpecifiedString("""{"col1":"+INF","col2":"+INF"}""",
+      TestRow(Float.PositiveInfinity, Double.PositiveInfinity))
+    testSpecifiedString("""{"col1":"+Infinity","col2":"+Infinity"}""",
+      TestRow(Float.PositiveInfinity, Double.PositiveInfinity))
+    testSpecifiedString("""{"col1":"Infinity","col2":"Infinity"}""",
+      TestRow(Float.PositiveInfinity, Double.PositiveInfinity))
+    testSpecifiedString("""{"col1":"-INF","col2":"-INF"}""",
+      TestRow(Float.NegativeInfinity, Double.NegativeInfinity))
+    testSpecifiedString("""{"col1":"-Infinity","col2":"-Infinity"}""",
+      TestRow(Float.NegativeInfinity, Double.NegativeInfinity))
+  }
```

Comment:
can we add some negative tests where the value is out of range? Wondering if we can copy the `from_json` suite in Spark.